src/bin/stratisd.rs

```diff
@@ -85,20 +85,44 @@ fn run() -> StratisResult<()> {
 
     let (dbus_conn, mut tree, dbus_context) = libstratis::dbus_api::connect(Rc::clone(&engine))?;
 
-    // Get a list of fds to poll for
-    let mut fds: Vec<_> = dbus_conn
-        .watch_fds()
-        .iter()
-        .map(|w| w.to_pollfd())
-        .collect();
+    let mut fds = Vec::new();
+
+    let mut eventable = engine.borrow_mut().get_eventable()?;
+    let poll_timeout = match eventable {
```

Comment:
You have to eventually make one list of `struct pollfd` because that's what `libc::poll` takes, see `man 2 poll`. And then that's also where you look after it returns to tell which (if any) of the fds have had an event. I'm not sure doing as you describe simplifies things -- so we now have an `Option<libc::pollfd>` but you still have to copy it into a Vec with the dbus fds, one after the other, and then check the array to dispatch events. How can this be made simpler? There seems to be a lot of irreducible ugliness here (which is what things like tokio is supposed to improve, and maybe we should use that in the future, but it has its own set of complexities that IMHO make this simpler right now.)